pkg/k8shandler/exec.go

```diff
@@ -0,0 +1,81 @@
+package k8shandler
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sclient"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+type ExecConfig struct {
+	namespace      string
+	labelName      string
+	labelPrefix    string
+	containerName  string
+	command        []string
+	kubeConfigPath string
+	masterURL      string
+}
+
+func PodExec(config *ExecConfig) (*bytes.Buffer, *bytes.Buffer, error) {
```

Comment:
Is there a reason we want to exec into the pod instead of using the REST endpoint to curl directly?  Are there advantages?

Reply:
We will also be using this if we're going to gate on the non-cluster service. We will need to have direct access to the pods.